sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -237,4 +245,42 @@ public void testNonDeterministicFunctionsAlwaysNew() throws Exception {
         assertThat(fn1, is(equalTo(fn2)));
         assertThat(fn1, is(not(equalTo(fn3))));
     }
+
+    @Test
+    public void testInPredicateWithSubqueryIsRewrittenToAnyEq() {
+        TestingDocTableInfoFactory docTableInfoFactory = new TestingDocTableInfoFactory(
+            ImmutableMap.of(
+                T3.T1_INFO.ident(), T3.T1_INFO,
+                T3.T2_INFO.ident(), T3.T2_INFO
+            )
+        );
+        Map<QualifiedName, AnalyzedRelation> sources = ImmutableMap.of(
+            new QualifiedName("t1"), T3.TR_1,
+            new QualifiedName("t2"), T3.TR_2
+        );
+        StatementAnalysisContext statementAnalysisContext = new StatementAnalysisContext(
+            SessionContext.SYSTEM_SESSION,
+            paramTypeHints,
+            Operation.READ,
+            null
+        );
+        Schemas schemas = new Schemas(
+            Settings.EMPTY,
+            ImmutableMap.of(
+                Schemas.DEFAULT_SCHEMA_NAME,
+                new DocSchemaInfo(Schemas.DEFAULT_SCHEMA_NAME, clusterService, functions, null, docTableInfoFactory)),
+            clusterService,
+            new DocSchemaInfoFactory(docTableInfoFactory, functions, null)
+        );
+        RelationAnalyzer analyzer = new RelationAnalyzer(clusterService, functions, schemas);
+        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(
+            functions,
+            SessionContext.SYSTEM_SESSION,
+            paramTypeHints,
+            new FullQualifiedNameFieldProvider(sources, ParentRelations.NO_PARENTS),
+            new SubqueryAnalyzer(analyzer, statementAnalysisContext)
+        );
+        Symbol symbol = expressionAnalyzer.convert(SqlParser.createExpression("t1.x in (select t2.y from t2)"), context);
+        assertThat(symbol, isSQL("(doc.t1.x = ANY(to_int_array(SelectSymbol{integer_table})))"));
```

Comment:
I had complained about it too and already extended the SqlExecutor so that it can be used to do this in a separate feature branch. I'd propose to merge this as is (without changing it here in this PR) and I can then do a follow up against master which introduces the SqlExecutor changes and cleans up this test